packages/apple-pay-integration/src/apple-pay-customer-strategy.ts

```diff
@@ -543,29 +543,39 @@ export default class ApplePayCustomerStrategy implements CustomerStrategy {
     }
 
     private async _getBraintreeDeviceData() {
-        const data = await this._braintreeIntegrationService.getDataCollector();
+        const braintreePaymentMethod = this._paymentIntegrationService
+            .getState()
+            .getPaymentMethod(ApplePayGatewayType.BRAINTREE);
 
-        return data.deviceData;
+        if (braintreePaymentMethod?.clientToken) {
+            const data = await this._braintreeIntegrationService.getDataCollector();
```

Comment:
Is there a way we can avoid BT calls?

Reply:
We can not avid it because we need to initialize BT to collect device data based on merchant configuration